packages/core/src/trpc.ts

```diff
@@ -33,10 +35,23 @@ export function trpcMiddleware(options: SentryTrpcMiddlewareOptions = {}) {
     };
 
     if (options.attachRpcInput !== undefined ? options.attachRpcInput : clientOptions && clientOptions.sendDefaultPii) {
-      trpcContext.input = normalize(rawInput);
-    }
+      if (rawInput !== undefined) {
+        trpcContext.input = normalize(rawInput);
+        setContext('trpc', trpcContext);
+      }
 
-    setContext('trpc', trpcContext);
+      if (getRawInput !== undefined && typeof getRawInput === 'function') {
+        getRawInput().then(
+          rawRes => {
+            trpcContext.input = normalize(rawRes);
+            setContext('trpc', trpcContext);
+          },
+          _e => {
+            // noop
+          },
+        );
```

Comment:
I think we may have a race condition here. We are only setting the context when getRawInput resolves, however that may happen after an error was captured 🤔 This could get tricky.

Also, I think we should probably grab the scope from outside of the `getRawInput()` async context and directly write the context there.

Reply:
yeah that's true, we have tested this on friday and it was working fine though – but we could generally just return an async middleware and await the payload if needed?